Packs/Okta/Integrations/OktaEventCollector/OktaEventCollector.py

```diff
@@ -81,34 +88,45 @@ def make_api_call(self):
         self.client.request.params.limit = str(limit_tmp - len(events))  # type: ignore
         return events
 
-    def _iter_events(self, last_object_ids: list) -> None:  # type: ignore  # pragma: no cover
+    def aggregated_results(self, last_object_ids: List[str] = None) -> List[dict]:  # pragma: no cover
         """
-        Function that responsible for the iteration over the events returned from the Okta api
+        Function to group the events returned from the api
         """
+        stored_events: list = []
+        max_events_to_fetch = int(self.client.request.params.limit)  # type: ignore
+        demisto.debug(f'max_events_to_fetch={max_events_to_fetch}')
+
+        while len(stored_events) < max_events_to_fetch:  # type: ignore
+            demisto.debug(f'stored_events={len(stored_events)}')
 
-        events: list = self.make_api_call()  # type: ignore
-        if last_object_ids:
-            events = GetEvents.remove_duplicates(events, last_object_ids)
-        while True:
-            yield events
-            last = events[-1]
-            self.client.set_next_run_filter(last['published'])
-            events: list = self.make_api_call()  # type: ignore
             try:
-                assert events
-            except (IndexError, AssertionError):
-                LOG('empty list, breaking')
+                events: list = self.make_api_call()  # type: ignore
+                demisto.debug(f'received {len(events)} number of events.')
+            except DemistoException as e:
+                if e.res is not None:
+                    if e.res.status_code == 429:
+                        demisto.debug('Got 429 status code, rate limit 0')
```

Comment:
once a 429 occurs will return stored_events but will not retry to fetch-events.
is this what we want ?

Reply:
yes. it is a simple implementation. Which basically assumes there is enough time and rate limit to fetch events. If we got rate limit now, we will have enough in the next run